Codegen/src/Nodes/TypedDataNode.cs

```diff
@@ -200,6 +205,48 @@ internal static string ExtractAnyCobolScalarTypeDef(ColumnsLayout? layout, TypeD
                         return "";
                     }
                     FlushConsumedTokens(layout, i - 1, customtype.CodeElement.ConsumedTokens, sb, out bHasPeriod);
+
+                    //Add any VALUE clause from dataDescriptionEntry
+                    if (bHasPeriod)
+                    {
+                        sb.Length--; //Trim trailing PeriodSeparator if any.
+                        bHasPeriod = false;
+                    }
+                    i = 0;
```

Comment:
Why to continue here if bHasPeriod is true?

Reply:
At this point we have the type info from the typedef but we still need to add the VALUE clause from the dataDef. It is also possible to use a tokenFilter to avoid outputting the PeriodSeparator bu I found it easier to simply remove it.

Reply:
But I mean if we have encountered a dot in any declaration, there is nothing else that can follow it.

Reply:
So I understand now that, first the PICTURE type tokens from the TypeDef are flushed followed by the any VALUE tokens.